activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -416,18 +416,24 @@ def supports_lazy_transactions?
         true
       end
 
+      def generate_advisory_lock_id
+        db_name_hash = Zlib.crc32(self.current_database)
+        schema_name_hash = Zlib.crc32(self.current_schema)
+        [db_name_hash, schema_name_hash].pack("L*").unpack("l*")
+      end
+
       def get_advisory_lock(lock_id) # :nodoc:
-        unless lock_id.is_a?(Integer) && lock_id.bit_length <= 63
-          raise(ArgumentError, "PostgreSQL requires advisory lock ids to be a signed 64 bit integer")
+        unless lock_id.is_a?(Array) && lock_id.all? { |id| id.is_a?(Integer) && id.bit_length <= 31 }
+          raise(ArgumentError, "PostgreSQL advisory lock should be an array of two signed 32 bit integers")
```

Comment:
The condition here allows for more then two items in the array, while the error message indicates exactly two.

It seems valuable to be explicit with the count, rather than allowing any size array.

Reply:
Thanks for your suggestions, @marksiemers!